azext_iot/tests/digitaltwins/test_dt_resource_lifecycle_int.py

```diff
@@ -141,7 +141,8 @@ def test_dt_resource(self):
         sleep(60)
 
         assert_common_resource_attributes(
-            self.wait_for_hostname(create_msi_output),
+            self.wait_for_hostname(
```

Comment:
We should consider retooling `wait_for_hostname` to not hide LRO issues (now that we know/confirmed the service behavior), so we can be confident the check_state callback is working, without additional wait loops https://github.com/vilit1/azure-iot-cli-extension/blob/dt_create2/azext_iot/tests/digitaltwins/__init__.py#L168

Reply:
Two wait calls will still be required - both are for identity assignments since the service can return a succeeded state without updating the identity